RocketChat/Rocket.Chat.ReactNative
# Chore: Migrate REST API - getUsernameSuggestion to Typescript

## Proposed changes

Migrate REST API - getUsernameSuggestion to Typescript

## Issue(s)	


## How to test or reproduce

- Create a new account in `open.rocket.chat` with special characters, like: `teste__/test1`
- Verify your email
- Then when you try to login again will popup a screen with title `Username`
- And the username inside the text input is the suggestion

## Screenshots

## Types of changes




## Checklist



## Further comments
